components/script/dom/htmlanchorelement.rs

```diff
@@ -581,21 +582,39 @@ pub fn follow_hyperlink(subject: &Element, hyperlink_suffix: Option<String>) {
     let source = document.browsing_context().unwrap();
 
     // Step 4-5: target attribute.
-    let target_attribute_value = subject.get_attribute(&ns!(), &local_name!("target"));
-
+    let subject_type = subject.get_attribute(&ns!(), &local_name!("target"));
+
+    let mut target_attribute_value = None;
+    if subject.is::<HTMLAreaElement>() || subject.is::<HTMLAnchorElement>() {
+        target_attribute_value = subject_type;
+
+        if !subject.has_attribute(&local_name!("target")) {
+            let base = (*document).base_element();
+            match base {
+                Some(base) => {
+                    let base_html_unwrap = (*base).upcast::<HTMLElement>().upcast::<Element>();
+                    if base_html_unwrap.has_attribute(&local_name!("target")) {
+                        target_attribute_value =
+                            base_html_unwrap.get_attribute(&ns!(), &local_name!("target"));
+                    } else {
+                        target_attribute_value = None;
+                    }
+                },
+                None => target_attribute_value = None,
+            };
+        }
+    }
```

Comment:
With a `get_element_target` function that returns `Option<DOMString>`, we could write this as:
```rust
if !(subject.is::<HTMLAreaElement>() || subject.is::<HTMLAnchorElement>() || subject.is::<HTMLFormElement>()) {
    return None;
}
if subject.has_attribute(&local_name!("target")) {
    return subject.get_attribute(&ns!(), &local_name!("target"));
}
document
    .base_element()
    .map(|base| base.upcast::<Element>())
    .and_then(|element| {
        if element.has_attribute(&local_name!("target")) {
            Some(element.get_attribute(&local_name!("target")))
        } else {
            None
        }
    })
```